### PR TITLE
Import native functions to improve opcache performance

### PR DESCRIPTION
**Pull request by LuomaJuha**
This pr imports native functions so opcache can turn them into opcode and improves loading times.

Here is some conversation about the topic: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/3048 and the rule added: https://cs.symfony.com/doc/rules/function_notation/native_function_invocation.html

### CODE REVIEW

**Comment by LuomaJuha** [comment]
Seems like small side-effects happened. Trying to adjust them at some point.

**Comment by demiankatz** [comment]
Thanks, @LuomaJuha. Is it possible that php-cs-fixer has a setting to insert use statements for functions instead of the leading backslashes? I think that might improve readability and reduce the number of long line issues created by the change.

**Comment by LuomaJuha** [comment]
@demiankatz     'global_namespace_import' => [
        'import_constants' => null,
        'import_functions' => true,
        'import_classes' => null,
    ], seems to do what I am looking for but for some reason it is not working. Gonna have to try and see why,

**Comment by demiankatz** [comment]
Thanks, @LuomaJuha -- that does look like it would be a better solution if you can get it to work. If you continue to have trouble with it, please let me know and I'll be happy to experiment on my end when time permits.

**Comment by LuomaJuha** [comment]
@demiankatz Hmm, seems like I have to backslash the functions before the global namespace import understands it, will try and push the new changes soon

**Comment by LuomaJuha** [comment]
Well, it almost worked like a charm but the use statements looks very awful. And non-class php files dont add use statements, but I think that is okay?

File: tests/vufind.php-cs-fixer.php
Comment (EreMaijala) [inline tests/vufind.php-cs-fixer.php]: Would it make sense to do this only for the optimizable functions? See https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v2.15.1/src/RuleSet.php#L194 and https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/4448.

Comment (LuomaJuha) [reply tests/vufind.php-cs-fixer.php]: Good idea, ill add and test that also

**Comment by demiankatz** [comment]
@LuomaJuha, it makes sense that the template files would not add use statements, since you only need a use statement to eliminate ambiguity when you declare a namespace, and none of the template files declare namespaces. Thus, I suspect they don't need any changes at all -- the function calls in those files MUST be the PHP-land versions, since there is no way to declare a user-land override without a namespace. (Though please correct me if I'm overlooking something!)

**Comment by LuomaJuha** [comment]
@demiankatz holy smokes, I managed to get it working as intended. I also updated the php-cs-fixer to higher version in this pr.

File: tests/vufind.php-cs-fixer.php
Comment (demiankatz) [inline tests/vufind.php-cs-fixer.php]: Since null is the default value for import_constants, would it be better to omit it here? Or do you think it might be worth changing to true in a subsequent PR?

File: tests/vufind.php-cs-fixer.php
Comment (demiankatz) [inline tests/vufind.php-cs-fixer.php:24]: I see that the default value for import_classes is true. I assume you're setting it to null here to avoid lots of unrelated changes getting mixed together -- which I think is a good idea! But should we consider setting this to true in a follow-up PR?

File: tests/vufind.php-cs-fixer.php
Comment (demiankatz) [inline tests/vufind.php-cs-fixer.php:56]: You changed ordered_imports here, but not in the template configuration below. I realize that this is probably because the setting is not particularly relevant to templates... but maybe it's better to make the configurations as consistent as possible so this doesn't confuse us in future if we compare the two files. Or is there a reason they need to be different?

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @LuomaJuha, this is great! I have a few small questions about the configuration settings and possible next steps, but as far as I can tell, all of the automatically generated changes here are correct and appropriate, and the full test suite is still passing. I'll be happy to approve and merge this once we finish making any minor adjustments to the configuration files  (which will likely have no immediate impact on any of the other files).

File: tests/vufind.php-cs-fixer.php
Comment (LuomaJuha) [reply tests/vufind.php-cs-fixer.php:24]: I think it would be a great followup

File: tests/vufind.php-cs-fixer.php
Comment (LuomaJuha) [reply tests/vufind.php-cs-fixer.php]: sure.

File: tests/vufind.php-cs-fixer.php
Comment (demiankatz) [reply tests/vufind.php-cs-fixer.php:24]: Great, please feel free to try it out after I merge this, and if you like the results, you can open a follow-up PR!

**Review by demiankatz — APPROVED** [review]
Looks good to me, and all tests still pass. Thanks again, @LuomaJuha!

**Pull request merged**
